src/main/scala/pipelined/cpu-bp.scala

```diff
@@ -55,12 +55,16 @@ class PipelinedCPUBP(implicit val conf: CPUConfig) extends BaseCPU {
     val readdata1 = UInt(32.W)
     val pc        = UInt(32.W)
     val pcplusfour= UInt(32.W)
-    val excontrol = new EXControl
-    val mcontrol  = new MControl
-    val wbcontrol = new WBControl
     val rs1       = UInt(5.W)    //pipelined only
     val rs2       = UInt(5.W)    //pipelined only
-    val branchpc  = UInt(32.W)
+    val branchpc  = UInt(32.W)   //bp only
+  }
+
+  // Control block of the IDEX register
+  class IDEXControl extends Bundle {
+    val ex = new EXControl
```

Comment:
These variables could have longer names (e.g., `ctrl_ex`)

The problem is when they're used below, it's hard to tell what's going on.

`id_ex_ctrl.io.in.ex.add`

What's `io`, `in`, `ex`? Everything is so short! I think `io` and `in`/`data` are fine, but more descriptive names for the control parts of the registers would be helpful.